database/db.go

```diff
@@ -64,16 +64,27 @@ func InsertVoters(voters []models.Voter) error {
 	mutex.Lock()
 	defer mutex.Unlock()
 
-	for _, voter := range voters {
-		var email string
-		err := db.QueryRow(`SELECT * FROM valid_voters WHERE email = $1`, voter).Scan(&email)
-		if err != sql.ErrNoRows {
-			continue
-		}
-		_, err = db.Exec(`INSERT INTO valid_voters VALUES ($1)`, voter)
-		if err != nil {
+	query := `SELECT * FROM valid_voters`
+	rows, err := db.Query(query)
+	if err != nil {
+		return err
+	}
+
+	alreadyAdded := map[models.Voter]bool{}
+	for rows.Next() {
+		var voter models.Voter
+		if err := rows.Scan(&voter); err != nil {
 			return err
 		}
+		alreadyAdded[voter] = true
+	}
+
+	for _, voter := range voters {
+		if !alreadyAdded[voter] {
+			if _, err = db.Exec(`INSERT INTO valid_voters VALUES ($1)`, voter); err != nil {
+				return err
```

Comment:
`db.Exec(`INSERT INTO valid_voters VALUES ($1)`, voter)`

Can the argument here be an array?
If so you could create some new voter array by using `voter.filter(voter => !alreadyAdded[voter])`.
Sorry for the js-syntax.

Reply:
No, I did not find a way to pass an array to insert in the `database/sql` package. This is what I tried to implement using variadics.

Reply:
Note that I've decided that I will not be able to finish this project in time for the initially planned deadline. Therefore I will take the time to properly reimplement the database using `gorm`, which will probably make this much nicer